src/NavierStokes.hpp

```diff
@@ -0,0 +1,355 @@
+    const double &deltat_,
+    const double &T_,
+    const unsigned int &step_)
+    : mesh_file_name(mesh_file_name_)
+    , mpi_size(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD))
+    , mpi_rank(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+    , pcout(std::cout, mpi_rank == 0)
+    , degree_velocity(degree_velocity_)
+    , degree_pressure(degree_pressure_)
+    , mesh(MPI_COMM_WORLD)
+    , deltat(deltat_)
+    , T(T_)
+    , step(step_)
+  {}
+
+  void setup();
+
+  void set_re_number(int Re);
+
+  void assemble_matrices();
+
+  void assemble_rhs(const double &time);
+
+  void assemble_static_matrices();
+
+  void solve_time_step(std::ostream &oss);
+
+  void assemble(const double &time);
+
+  void output(const unsigned int &time_step) const;
+
+  void solve(unsigned int time_step = 0);
+
+  void export_data(const unsigned int &time_step);
+
+  void compute_ordered_dofs_indices();
+
+  void import_data(const unsigned int &time_step);
+
+  void post_process(const unsigned int &initial_time_step,
+    const unsigned int &final_time_step, const unsigned int &step);
+
+  void compute_forces(const double &time);
+
+
+protected:
+  const std::string mesh_file_name;
+
+  const unsigned int mpi_size;
+  const unsigned int mpi_rank;
+  ConditionalOStream pcout;
+
+  const unsigned int degree_velocity;
+  const unsigned int degree_pressure;
+
+
+  parallel::fullydistributed::Triangulation<dim> mesh;
+  std::unique_ptr<FiniteElement<dim>> fe;
+  std::unique_ptr<Quadrature<dim>> quadrature;
+  std::unique_ptr<Quadrature<dim - 1>> quadrature_face;
+
+  DoFHandler<dim> dof_handler;
+
+  IndexSet locally_owned_dofs;
+  std::vector<IndexSet> block_owned_dofs;
+
+  IndexSet locally_relevant_dofs;
+  std::vector<IndexSet> block_relevant_dofs;
+
+  // Useful for saving the renumbered indexes
+  std::vector<types::global_dof_index> renumbered_dofs;
+
+  TrilinosWrappers::BlockSparseMatrix system_matrix;
+  TrilinosWrappers::MPI::BlockVector deltat_lumped_mass_inv;
+  TrilinosWrappers::MPI::BlockVector system_rhs;
+  TrilinosWrappers::MPI::BlockVector solution_owned;
+  TrilinosWrappers::MPI::BlockVector solution;
+
+  double nu = 1.e-3;
+  const double p_out = 10.0;
+  const double Diameter = 0.1;
+
+  const double deltat;
+
+  double time;
+  const double T;
+  const unsigned int step;
+
+  ForcingTerm forcing_term;
+  InletVelocity inlet_velocity;
+  InitialConditions initial_conditions;
+  //FunctionH function_h;
+
+  double drag, lift;
+  double cd, cl;
+};
+
+
+class NavierStokes::PreconditionIdentity
+{
+public:
+// Application of the preconditioner: we just copy the input vector (src)
+// into the output vector (dst).
+  void
+  vmult(TrilinosWrappers::MPI::BlockVector &dst,
+    const TrilinosWrappers::MPI::BlockVector &src) const
+  {
+    dst = src;
+  }
+
+protected:
+};
+
+
+class NavierStokes::PreconditionASIMPLE
+{
+public:
+  // Application of the preconditioner: we just copy the input vector (src)
+  // into the output vector (dst).
```

Comment:
I think this comment is a left-over from copypasting.

Remember that "incorrect documentation is worse than no documentation".